src/core/overflowsafe_type.hpp

```diff
@@ -34,7 +34,7 @@ class OverflowSafeInt
 
 	inline OverflowSafeInt& operator = (const OverflowSafeInt& other) { this->m_value = other.m_value; return *this; }
 
-	inline OverflowSafeInt operator - () const { return OverflowSafeInt(-this->m_value); }
+	inline OverflowSafeInt operator - () const { return OverflowSafeInt(this->m_value == T_MIN ? T_MAX : -this->m_value); }
```

Comment:
Re commit message `Fix: OverflowSafeInt negation not handling INT64_MIN `
Here too the commit message could use a bit more detail what is going on, for future us. Something like:

```
Fix: OverflowSafeInt negation not handling INT64_MIN

INT64_MIN negated is above INT64_MAX, and would overflow.
Instead, when negating INT64_MIN make it INT64_MAX.
This does mean that -(-(INT64_MIN)) != INT64_MIN.
```

Or something along those lines